### PR TITLE
lanetix/issues#2729 - Upgrade to latest eslint (v0.24)

### PR DESCRIPTION
**Pull request by OronNadiv**
@lanetix/engineers CR Please.  Thanks.
lanetix/issues#2729 - Upgrade to latest eslint (v0.24)

### CODE REVIEW

File: test/spec.js [hunk 1]
```diff
@@ -1,16 +1,18 @@
+'use strict';
+
+/* eslint "no-unused-expressions": 0 */
```
Comment (tone81) [inline test/spec.js]: How many violations would we get if we re-enable this check?

**Comment by tone81** [comment]
One question; otherwise, :goberserk:

**Comment by armw4** [comment]
Thought we we going away from bin scripts and sticking to npm run blah instead.

**Comment by scarney81** [comment]
`npm` by default exposes `node_modules/.bin` in the `PATH`. We should definitely err on the side of putting it in the package.json file as opposed to a bin script. If we _have_ to create a bin script for some reason, then it should be a `.js` file where possible so that it can be linted.

**Comment by armw4** [comment]
There's also already a test target inside package.json that uses gulp which makes bin/test all the more redundant.

File: test/spec.js (hunk 1 above)
Comment (OronNadiv) [inline test/spec.js]: Too many (30-40).  Every time we use ```redirect.called.should.be.???```  it will show eslint error.

**Comment by OronNadiv** [comment]
Good comments @armw4 and @scarney81.  I'll remove the bin folder if possible.

**Pull request merged**